packages/flutter_tools/lib/src/isolated/native_assets/targets.dart

```diff
@@ -0,0 +1,402 @@
+    Architecture architecture,
+  ) {
+    return <AssetBuildTarget>[
+      WindowsAssetTarget(architecture: architecture, supportedAssetTypes: supportedAssetTypes),
+    ];
+  }
+
+  static List<MacOSAssetTarget> _macTargets(
+    Map<String, String> environmentDefines,
+    List<SupportedAssetTypes> supportedAssetTypes,
+  ) {
+    return getDarwinArchsFromEnv(environmentDefines)
+        .map(getNativeMacOSArchitecture)
+        .map(
+          (Architecture architecture) => MacOSAssetTarget(
+            architecture: architecture,
+            supportedAssetTypes: supportedAssetTypes,
+          ),
+        )
+        .toList();
+  }
+
+  static List<AndroidAssetTarget> _androidTargets(
+    TargetPlatform targetPlatform,
+    Map<String, String> environmentDefines,
+    List<SupportedAssetTypes> supportedAssetTypes,
+  ) {
+    return _androidArchs(targetPlatform, environmentDefines[kAndroidArchs])
+        .map(getNativeAndroidArchitecture)
+        .map(
+          (Architecture architecture) => AndroidAssetTarget(
+            architecture: architecture,
+            supportedAssetTypes: supportedAssetTypes,
+            environmentDefines: environmentDefines,
+          ),
+        )
+        .toList();
+  }
+
+  static List<IOSAssetTarget> _iosTargets(
+    Map<String, String> environmentDefines,
+    FileSystem fileSystem,
+    List<SupportedAssetTypes> supportedAssetTypes,
+  ) {
+    final List<DarwinArch> iosArchitectures =
+        _emptyToNull(environmentDefines[kIosArchs])?.split(' ').map(getIOSArchForName).toList() ??
+        <DarwinArch>[DarwinArch.arm64];
+    return iosArchitectures
+        .map(getNativeIOSArchitecture)
+        .map(
+          (Architecture architecture) => IOSAssetTarget(
+            environmentDefines: environmentDefines,
+            fileSystem: fileSystem,
+            architecture: architecture,
+            supportedAssetTypes: supportedAssetTypes,
+          ),
+        )
+        .toList();
+  }
+
+  static List<AssetBuildTarget> _webTarget(List<SupportedAssetTypes> supportedAssetTypes) =>
+      <AssetBuildTarget>[WebAssetTarget(supportedAssetTypes: supportedAssetTypes)];
+
+  static List<AssetBuildTarget> _flutterTesterTarget(
+    List<SupportedAssetTypes> supportedAssetTypes,
+  ) {
+    return <AssetBuildTarget>[FlutterTesterAssetTarget(supportedAssetTypes: supportedAssetTypes)];
+  }
+}
+
+final class WebAssetTarget extends AssetBuildTarget {
+  WebAssetTarget({required super.supportedAssetTypes});
+
+  @override
+  List<ProtocolExtension> get extensions => <ProtocolExtension>[...dataAssetExtensions];
+
+  @override
+  String get targetString => 'web';
+}
+
+sealed class CodeAssetTarget extends AssetBuildTarget {
+  CodeAssetTarget({
+    required super.supportedAssetTypes,
+    required this.architecture,
+    required this.os,
+  });
+
+  final Architecture architecture;
+  final OS os;
+
+  late final CCompilerConfig? cCompilerConfigSync;
```

Comment:
This and `setCCompilerConfig()` need documentation since it's not obvious what they're for.